src/core/document.js

```diff
@@ -993,7 +993,23 @@ class PDFDocument {
     promises.length = 0;
     pdfFonts.length = 0;
 
+    const i = missingFonts.indexOf("MyriadPro");
+    const hasMyriad = i !== -1;
+    if (hasMyriad) {
+      missingFonts.splice(i, 1);
+    }
+    missingFonts.push("MyriadPro");
+    const reallyMissingFonts = new Set();
+
     for (const missing of missingFonts) {
+      if (
+        missing === "MyriadPro" &&
+        !hasMyriad &&
+        reallyMissingFonts.size === 0
+      ) {
+        break;
+      }
+
```

Comment:
I'm finding this, and the new code above, slightly confusing to reason about; any chance that it could be simplified (or re-factored) somewhat?

Reply:
The idea was to load Myriad only if Myriad is present in the list of the missing fonts (=> hasMyriad) or if a font has no substitution (.size === 0) and for the latter we need to review all the missing fonts so it's why I put the MyriadPro at the of the list.
Anyway you're right: I'm pretty sure that I'll find myself pretty painful to read this code in 2 weeks...

Reply:
Thank you; the updated patch is *a lot* clearer and easier to follow!